### PR TITLE
libopae: add  user clock API

### PR DESCRIPTION
**Pull request by anandaravuri**

### CODE REVIEW

File: common/include/opae/userclk.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Copyright(c) 2017, Intel Corporation
```
Comment (tswhison) [inline common/include/opae/userclk.h]: 2018 :)

File: tools/extra/userclk/main.c [hunk 2]
```diff
@@ -102,26 +101,28 @@ void print_err(const char *s, fpga_result res)
...
+
+	fpga_handle        accelerator_handle;
 
 	// Parse command line
 	if ( argc < 2 ) {
 		UserClkAppShowHelp();
 	return 1;
 	} else if ( 0!= ParseCmds(&userclkCmdLine, argc, argv) ) {
-		FPGA_ERR( "Error scanning command line \n.");
+	fprintf(stderr, "Error scanning command line \n");
```
Comment (tswhison) [inline tools/extra/userclk/main.c]: Just fyi, you could use OPAE_ERR.

File: libopae/plugins/xfpga/userclk.c [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	}
+
+	p = strstr(_token->sysfspath, FPGA_SYSFS_AFU);
+	if (NULL == p) {
+		FPGA_ERR("Invalid sysfspath in token");
+		return FPGA_INVALID_PARAM;
+	}
+
+	p = strrchr(_token->sysfspath, '.');
+	if (NULL == p) {
```
Comment (r-rojo) [inline libopae/plugins/xfpga/userclk.c]: same comment here about magic/objtype (ACCELERATOR)

File: libopae/plugins/xfpga/xfpga.h [hunk 4]
```diff
@@ -187,6 +187,13 @@ fpga_result xfpga_fpgaDestroyObject(fpga_object *obj);
 fpga_result xfpga_fpgaObjectRead(fpga_object obj, uint8_t *buffer,
 				 size_t offset, size_t len, int flags);
 
+fpga_result xfpga_fpgaSetUserClock(fpga_handle handle,
+							uint64_t low_clk, uint64_t high_clk, int flags);
```
Comment (r-rojo) [inline libopae/plugins/xfpga/xfpga.h]: indentation

File: libopae/plugins/xfpga/xfpga.h [hunk 5]
```diff
@@ -187,6 +187,13 @@ fpga_result xfpga_fpgaDestroyObject(fpga_object *obj);
...
+
+fpga_result xfpga_fpgaGetUserClock(fpga_handle handle,
+							uint64_t *low_clk, uint64_t *high_clk, int flags);
```
Comment (r-rojo) [inline libopae/plugins/xfpga/xfpga.h]: indentation

File: tools/extra/userclk/main.c [hunk 6]
```diff
@@ -102,26 +101,28 @@ void print_err(const char *s, fpga_result res)
...
 
-fpga_result get_fpga_port_sysfs(fpga_token token,char* sysfs_port,int portid);
 int ParseCmds(struct UserClkCommandLine *userclkCmdLine, int argc, char *argv[]);
 
 int main( int argc, char** argv )
 {
 	fpga_properties filter             = NULL;
 	uint32_t num_matches               = 1;
-	char sysfs_path[SYSFS_PATH_MAX]    = {0};
+//	char sysfs_path[SYSFS_PATH_MAX]    = {0};
```
Comment (r-rojo) [inline tools/extra/userclk/main.c]: remove dead code

**Pull request merged**